VACMS-6812 Create audit tool for rich text fields

msbtterswrth opened the pull request:
## Description

Closes #6812 .

## Testing done
navigated to new view and followed qa steps

## Screenshots
![image](https://user-images.githubusercontent.com/8375335/139128845-a47de880-0b71-4513-91a4-403d535593ba.png)

## QA steps

As cms editor I can
1. navigate to /admin/content/audit/rich-text-field-audit and view a report
2. filter by content type
3. search for content within the rich text fields
4. export results to csv
- [x] Users can filter by content type
- [x] Users can see the output of long text, formatted fields in the report
- [x] Users can export the report
- [x] update url to admin/content/audit/{{name}}
- [x] update perms for content admin and admin

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change.
  - [x] Merge & carry on unburdened by announcements

msbtterswrth commented on config/sync/views.view.rich_text_field_audit.yml:
```diff
@@ -0,0 +1,855 @@
+        - 'config:field.storage.node.field_full_story'
+        - 'config:field.storage.node.field_press_release_blurb'
+        - 'config:field.storage.node.field_press_release_fulltext'
+  page_1:
```
updated!

msbtterswrth commented on config/sync/views.view.rich_text_field_audit.yml:
```diff
@@ -0,0 +1,855 @@
+    position: 1
+    display_options:
+      display_extenders: {  }
+      path: rich-text-field-audit
```
updated to /admin/content/audit/rich-text-field-audit

msbtterswrth commented on config/sync/views.view.rich_text_field_audit.yml:
```diff
@@ -0,0 +1,855 @@
+          caption: ''
+          summary: ''
+          description: ''
+          columns:
```
added!

msbtterswrth commented on config/sync/views.view.rich_text_field_audit.yml:
```diff
@@ -0,0 +1,855 @@
+          entity_type: node
+          entity_field: status
+          id: status
+          expose:
```
also added!

swirtSJW reviewed (commented):
This looking pretty good overall:
Just a couple things.  

- Update the paths in the QA steps.   So we don't need to dig through the config to figure out what and where to go.
- We have some existing patterns in terms of column order and labels.  It would be good to align yours to existing patterns
![image](https://user-images.githubusercontent.com/5752113/139722946-ad8689ef-0aa5-4c36-ba55-c6f0deb4c266.png)

msbtterswrth commented:
Ok i've updated the order of the view columns, I didn't know there was a specific order, but it makes me happy that there is one :) Additionally I've added the last changed date to match the view you send as a guide. 

I also fixed the link in the qa steps, thank you for catching that.

swirtSJW commented:
> I didn't know there was a specific order, but it makes me happy that there is one :)
Yah it is not so much that there is a specific order as some columns will not exist on some Views, it is more about trying to have our tools have consistent feel across them.  
Things like: 

- order of columns
- naming of columns
- column sorts
- date formats
- filter order
- filter names
- edit buttons

should all provide as consistent a look and feel as possible.


Often we try to call out an existing View that is similar to model a new View upon in our refinement or planning sessions but, this one kind of bi-passed that process.

swirtSJW commented:
@msbtterswrth I just pushed up a commit to address the other items in my previous comment.
![image](https://user-images.githubusercontent.com/5752113/140082018-90dce8bf-360c-491e-b247-0a748d76a43d.png)

swirtSJW reviewed (approved):
Approved.   Just waiting on the last test run to complete.
